ubuntu_image/__main__.py

```diff
@@ -181,7 +337,14 @@ def parseargs(argv=None):
     return args
 
 
+argparse.ArgumentParser.is_subcommand_exist = is_subcommand_exist
+argparse.ArgumentParser.get_modified_arguments = get_modified_arguments
```

Comment:
I am not a fan of such 'hacking in' of methods to already defined classes. Seeing how those are used, these two functions could easily be made local functions in __main__ and just take all the required arguments as parameters, without relying on self. Yes, we'd have to access some private elements, but it's not explicitly disallowed. That way it will be easily unit-testable and much more readable IMO. So: I'd remove these two lines, modify both to accept a 'subparsers' parameter and use them like that.